### PR TITLE
[Fix #9608] Fix a false positive for `Layout/EmptyLineAfterGuardClause`

### LINKED ISSUE #9608: Make Layout/EmptyLineAfterGuardClause accept RuboCop directive after guard clause
## Expected behavior

https://github.com/rubocop/rubocop/issues/9146, but I can reproduce it with rubocop-1.11.0 with config:
```
require:
  - rubocop-performance

AllCops:
  NewCops: enable
  TargetRubyVersion: 3.0

Style/FrozenStringLiteralComment:
  Enabled: false
```
and code:
```
def xxx
  # rubocop:disable Performance/InefficientHashSearch
  # `rule` is a `Dry::Validation::Rule`, not a `Hash`
  return true if rule.keys.include?(@key_chain)
  # rubocop:enable Performance/InefficientHashSearch

  false
end
```

## Actual behavior

```
$ rubocop --debug /tmp/tst.rb
For /tmp: configuration from /tmp/.rubocop.yml
configuration from /home/ojab/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-performance-1.10.1/config/default.yml
configuration from /home/ojab/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-performance-1.10.1/config/default.yml
Default configuration from /home/ojab/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.11.0/config/default.yml
Inspecting 1 file
Scanning /tmp/tst.rb
C

Offenses:

tst.rb:4:3: C: [Correctable] Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
  return true if rule.keys.include?(@key_chain)
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

## Steps to reproduce the problem

Run the code above with config above.

## RuboCop version
```
$ rubocop -V
1.11.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-linux)
  - rubocop-performance 1.10.1
```

### PR DESCRIPTION
**Pull request by koic**
Fixes #9608.

This PR fixes a false positive for `Layout/EmptyLineAfterGuardClause` when using guard clause is after `rubocop:enable` comment.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
I'm assuming that's problem existing in other cops checking for a blank line after something.

File: lib/rubocop/cop/layout/empty_line_after_guard_clause.rb [hunk 1]
```diff
@@ -41,20 +41,21 @@ class EmptyLineAfterGuardClause < Base
 
         MSG = 'Add empty line after guard clause.'
         END_OF_HEREDOC_LINE = 1
+        RUBOCOP_ENABLE_COMMENT_PATTERN = /#\s*rubocop\s*:\s*enable/.freeze
```
Comment (AndreiEres) [inline lib/rubocop/cop/layout/empty_line_after_guard_clause.rb:44]: We can use DirectiveComment here

File: lib/rubocop/cop/layout/empty_line_after_guard_clause.rb [hunk 2]
```diff
@@ -85,8 +86,10 @@ def contains_guard_clause?(node)
           node.if_branch&.guard_clause?
         end
 
-        def next_line_empty?(line)
-          processed_source[line].blank?
+        def next_line_empty_or_rubocop_enable_comment?(line)
```
Comment (AndreiEres) [inline lib/rubocop/cop/layout/empty_line_after_guard_clause.rb:89]: I think we have to still hold empty line after directive comment. What do you think?

**Comment by AndreiEres** [comment]
> I'm assuming that's problem existing in other cops checking for a blank line after something.

If it is a general behaviour, maybe we can add an extra helper class to check it?

File: lib/rubocop/cop/layout/empty_line_after_guard_clause.rb (hunk 2 above)
Comment (bbatsov) [reply lib/rubocop/cop/layout/empty_line_after_guard_clause.rb:89]: Yeah, I agree. There should still be an empty line, but it should come right after the enable directive.

File: spec/rubocop/cop/layout/empty_line_after_guard_clause_spec.rb [hunk 3]
```diff
@@ -199,6 +199,28 @@ def foo
     RUBY
   end
 
+  it 'registers and corrects when using guard clause is after `rubocop:disable` comment' do
```
Comment (bbatsov) [inline spec/rubocop/cop/layout/empty_line_after_guard_clause_spec.rb:202]: Theres should also be a test making sure an empty line is required after the enable directive.

**Comment by dvandersluis** [comment]
Should this be specific to directive comments, or maybe generalized to any comment (or multiple lines of comment) after the clause?

**Comment by bbatsov** [comment]
@koic ping :-)

**Comment by koic** [comment]
Thank you for reviews. I updated this PR and I will proceed separately for similar other cops.

**Comment by bbatsov** [comment]
Thanks!
